prescient.el

```diff
@@ -358,7 +358,9 @@ enclose literal substrings with capture groups."
           (pcase method
             (`literal
              (prescient--with-group
-              (regexp-quote subquery)
+              (if (eq search-default-mode #'char-fold-to-regexp)
+                  (char-fold-to-regexp subquery)
+                (regexp-quote subquery))
```

Comment:
I would like this to be enabled unconditionally, not dependent on `search-default-mode`. We don't actually support `search-default-mode` properly, and conditioning this behavior on a particular value for `search-default-mode` misleadingly suggests that we also support other values.

Reply:
Oh, my bad. Wanted to make it unconditional, but instead... Pushed the change.